lib/paquito/active_record_coder.rb

```diff
@@ -99,18 +102,25 @@ def attributes_for_database(record)
         attributes
       end
 
-      def deserialize_record(class_name, attributes_from_database, new_record = false)
+      def deserialize_record(class_name, attributes_from_database, new_record = false, hash = nil)
         begin
           klass = Object.const_get(class_name)
         rescue NameError
           raise ClassMissingError, "undefined class: #{class_name}"
         end
 
+        raise ColumnsHashMismatch if hash && (hash != columns_hash(klass))
+
         # Ideally we'd like to call `klass.instantiate`, however it doesn't allow to pass
         # wether the record was persisted or not.
         attributes = klass.attributes_builder.build_from_database(attributes_from_database, EMPTY_HASH)
         klass.allocate.init_with_attributes(attributes, new_record)
       end
+
+      def columns_hash(klass)
+        str = klass.columns_hash.map { |name, column| [name, column.sql_type].join(":") }.join("|")
```

Comment:
This generates a string like:

```
id:INTEGER|name:varchar|settings:BLOB|owner_id:INTEGER
```

Is this enough? 🤔

Reply:
I went with the first 6 characters to save space, so `f480f6` here. That's a space of `128**6`, or `4398046511104`, so collisions should be very unlikely.

Reply:
I assume `sql_type` is the right choice here but not 100% sure. 🤔